### PR TITLE
Support for AWS Inferentia instances on batch

### PR DESCRIPTION
**Pull request by DanCorvesor**
As per this documentation explaining how to set up Inferentia ECS instances, this PR adds support and specifically a new batch inferentia parameter to do this [Working with inference workloads on Amazon ECS - Amazon Elastic Container Service](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-inference.html).

Further details about this in Slack thread https://outerbounds-community.slack.com/archives/C020U025QJK/p1647436142654879.

### CODE REVIEW

**Comment by savingoyal** [comment]
@DanCorvesor Have you tested this on Step Functions?

**Comment by savingoyal** [comment]
f-strings are not supported in Python 3.5 causing the tests to fail.

File: metaflow/plugins/aws/batch/batch_cli.py [hunk 1]
```diff
@@ -140,6 +140,7 @@ def kill(ctx, run_id, user, my_runs):
 @click.option("--shared-memory", help="Shared Memory requirement for AWS Batch.")
 @click.option("--max-swap", help="Max Swap requirement for AWS Batch.")
 @click.option("--swappiness", help="Swappiness requirement for AWS Batch.")
+@click.option("--inferentia", help="Inferentia requirement for AWS Batch.")
```
Comment (savingoyal) [inline metaflow/plugins/aws/batch/batch_cli.py:143]: Is my understanding correct that the Batch Compute Environment needs to be entirely composed of Inf1 instances to have any guarantee that the job will launch on top of the inf1 instance?

**Review by savingoyal — COMMENTED** [review]
@DanCorvesor The UX looks good to me. Just pointed a few small amends - we should be able to get this PR merged after those are addressed.

**Comment by DanCorvesor** [comment]
> @DanCorvesor Have you tested this on Step Functions?

No, we don't use Step Functions at all hence why I haven't - are you able to test this for me?

File: metaflow/plugins/aws/batch/batch_cli.py (hunk 1 above)
Comment (DanCorvesor) [reply metaflow/plugins/aws/batch/batch_cli.py:143]: Yes, that is my understanding also

**Comment by DanCorvesor** [comment]
@savingoyal Are you able to help me work out why the python 3.5 checks are failing?

File: metaflow/plugins/aws/batch/batch_client.py [hunk 2]
```diff
@@ -247,6 +248,24 @@ def _register_job_definition(
                         "maxSwap"
                     ] = int(max_swap)
 
+        if inferentia:
```
[the remark this answers is not in the document]
Comment (DanCorvesor) [reply metaflow/plugins/aws/batch/batch_client.py:251]: I think it should match the behaviour of gpu=0 which wouldn't schedule the job at all, I have addressed this

**Comment by mattmcclean** [comment]
Any update on this PR being merged ?

**Comment by DanCorvesor** [comment]
> Any update on this PR being merged ?

I am going to chase this today

**Pull request closed without merging**